src/Controller/UserController.php

```diff
@@ -135,8 +158,12 @@ public function createUserAction(
 
         $encoded = $encoder->encodePassword($newUser, $newUser->getPassword());
         $newUser->setPassword($encoded);
+        if ($newUser->getRoles() === ["ROLE_USER"]) {
+            $newUser->setRoles(["ROLE_USER"]);
+        }
```

Comment:
seems to be useless? Why do you reassign what we get in from the getter?

Reply:
Because the "ROLE_USER" is returned by `getRoles()` even if the user does not have it in the database, so I set it manually to be sure it will be saved and so correctly used by the API.

Reply:
So what do I do? Should I leave as it is or change the getter like that:
```php
    public function getRoles(): array
    {
        // guarantee every user at least has ROLE_USER
        if (empty($this->roles)) {
            $this->setRoles([self::ROLE_USER]);
            
            return [self::ROLE_USER];
        }
        
        return $this->roles;
    }
```